consensus/src/block_storage/sync_manager.rs

```diff
@@ -266,6 +266,8 @@ impl BlockStore {
         if !self.need_sync_for_ledger_info(highest_commit_cert.ledger_info()) {
             return Ok(());
         }
+
+        let window_size = self.inner.read().get_window_size();
```

Comment:
@hariria  It does force you to lock for this. But guess it shouldn't be too expensive?